### PR TITLE
Add secrets api to convenient layer

### PR DESCRIPTION
**Pull request by pruthvidhodda**
# Description

Added secrets api to convenient layer

## Issue reference

We strive to have all PR being opened based on an issue, where the problem or feature have been discussed prior to implementation.

Please reference the issue this PR will close: #76 

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [x] Created/updated tests
* [x] Extended the documentation

### CODE REVIEW

File: dapr/clients/grpc/client.py [hunk 1]
```diff
@@ -282,3 +283,62 @@ def publish_event(
         return DaprResponse(
             headers=call.initial_metadata(),
             trailers=call.trailing_metadata())
+
+    def get_secret(
+            self,
+            store_name: str,
+            key: str,
+            metadata: Optional[MetadataTuple] = ()) -> GetSecretResponse:
```
Comment (youngbupark) [inline dapr/clients/grpc/client.py:291]: this metadata is for grpc metadata. 

For secret store metadata, please add new arg - secret_metadata: Dict[str, str]

```
   def get_secret(
            self,
            store_name: str,
            key: str,
            secret_metadata: Optional[Dict[str, str]] = {},
            metadata: Optional[MetadataTuple] = ()) -> GetSecretResponse:
```

File: dapr/clients/grpc/client.py [hunk 2]
```diff
@@ -282,3 +283,62 @@ def publish_event(
...
+        Returns:
+            :class:`GetSecretResponse` object with the secret and metadata returned from callee
+        """
+
+        metadata_dict = dict()
+
+        for item in metadata:   # type: ignore
+            if not isinstance(item[1], str):
+                raise ValueError(f'invalid metadata value type {type(item[1])}')
+            metadata_dict[str(item[0])] = str(item[1])
```
Comment (youngbupark) [inline dapr/clients/grpc/client.py:332]: once you have `secretstore_metadata` arg, you do not need this code.

File: dapr/clients/grpc/client.py [hunk 3]
```diff
@@ -282,3 +283,62 @@ def publish_event(
...
+
+        req = api_v1.GetSecretRequest(
+            store_name=store_name,
+            key=key,
+            metadata=metadata_dict)
+
+        response, call = self._stub.GetSecret.with_call(req)
```
Comment (youngbupark) [inline dapr/clients/grpc/client.py:339]: please pass `metadata`

File: dapr/clients/grpc/client.py (hunk 1 above)
Comment (pruthvidhodda) [reply dapr/clients/grpc/client.py:292]: I was confused b/w them. Changed it.

**Pull request merged**